packages/template-retail-react-app/app/commerce-api/einstein.js

```diff
@@ -102,7 +102,7 @@ class EinsteinAPI {
         const endpoint = `/activities/${this.config.siteId}/viewSearch`
         const method = 'POST'
 
-        const products = searchResults.hits.map((product) => {
+        const products = searchResults?.hits?.map((product) => {
```

Comment:
just double checking here.

Before the change, `products` is always an array.

After the change `products` can be an array or undefined. Does the code in this module handles the undefined situation?

Reply:
At that point is on the API to handle the error. 
I'm guessing some APIs will error out if we send products undefined and some will accept that value as undefined.

We can also make the decision of not sending requests to the API when `products` is undefined and handle the error shown to users.

The `viewCategory` endpoint returns a 400 error when we send `products = undefined`:
```
POST http://localhost:3000/mobify/proxy/einstein/v3/activities/aaij-MobileFirst/viewCategory 400 (Bad Request)

{"error":{"code":"ValidationError","msg":"There was a problem validating the request. See `errors`.","errors":[{"msg":"Must provide a `products` array.","param":"products","location":"body"}]}}
```

That's why I added the guard to the PLP.

I don't see any validation on our side for the Einstein endpoint, so for simplicity and consistency, I'd leave it as it is now.